server.js

```diff
@@ -18,9 +18,121 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send({
+    responseMessage: "Happy Thoughts API by Yu Miao",
+    endpoints:
+    {
+      "/thoughts":"display all happy thoughts max 20",
+      "/thoughts/:thoughtId/like":"like a post by inreasing the hearts by 1",
+    }
+  });
 });
 
+//thoughs model
+const { Schema } = mongoose;
+const ThoughtSchema = new Schema ({
+  message : {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140,
+    trim: true
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type : Date,
+    default: new Date()
+  }
+});
+
+const ThoughtList = mongoose.model("Thought", ThoughtSchema)
+// Endpoints 1 : GET/thoughts, all happy thoughts max 20
+app.get("/thoughts", async(req,res) => {
+  try {
+    const thoughts = await ThoughtList.find().sort({createdAt:"desc"}).limit(20).exec()
+    res.status(201).json({
+      success: true,
+      response: thoughts,
+      message: "created successfully"
+    })
+
+  } catch (e) {
+    res.status(400).json({
+      success: false,
+      response: e,
+      message: "error occured"
+    });
+
+  }
+
+})
+// Endpoint 2: POST/thoughts, create a happy thoughts
+app.post("/thoughts", async(req,res)=>{
+  const { message } = req.body
+  try {
+  const newThought = await new ThoughtList({message}).save()
+    res.status(201).json({
+      success: true,
+      response: newThought,
+      message: "post successfully"
+    })
+
+  } catch (e) {
+    res.status(400).json({
+      success: false,
+      response: e,
+      message: "can not post your thought"
+    });
+  }
+
+})
```

Comment:
The update function is very similar but could have been good to implement for practice purposes, but this way works fine aswell!